docs/tutorial/spark-tutorial/t-spark-streaming.md

```diff
@@ -0,0 +1,138 @@
+```
+
+Now, let's deploy Zookeeper and Kafka using the `zookeeper-k8s` and `kafka-k8s` charms. Single units of these charm should be enough.
+
+```bash
+juju deploy zookeeper-k8s --series=jammy --channel=edge
+juju deploy kafka-k8s --series=jammy --channel=edge
+```
+
+The current status of the Juju model can be seen using the command:
+```bash
+juju status --watch 1s
+```
+
+Wait until the two charms are deployed and are in active and idle state. Now, for Kafka to be able to connect to Zookeeper, we need to integrate these two charms. Run the following command to integrate them.
+
+```bash
+juju integrate kafka-k8s zookeeper-k8s
+```
+
+For us to experiment with the streaming feature, we would want some sample streaming data to be generated continuously in real time. For that, we can use the `kafka-test-app` charm as a producer of events. Deploy the charm, and integrate it with `kafka-k8s` so that it is able to write messages to Kafka.
+
+```bash
+juju deploy kafka-test-app --series=jammy --channel=edge --config role=producer --config topic_name=spark-streaming-store --config num_messages=1000
+
+juju integrate kafka-test-app kafka-k8s
+```
+
+Now the messages will be generated and written to Kafka. But in order to consume these messages, we need the credentials to establish the connection between Spark and Kafka. For the retreival of credentials, we are going to use the `data-integrator` charm. Deploy the charm and integrate it with `kafka-k8s` with the following commands:
+
+```bash
+juju deploy data-integrator --series=jammy --channel=edge --config extra-user-roles=consumer,admin --config topic-name=spark-streaming-store
+
+juju integrate data-integrator kafka-k8s 
+```
+
+Once the charm is deployed and integrated, we can get the credentials to connect to Kafka by running the `get-credentials` action exposed by the `data-integrator` charm.
+
+```bash
+juju run-action data-integrator/0 get-credentials --wait 
+```
+
+You can view the credentials such as `username` and `password` in the output. These can be stored as variables using the following commands:
+
+```bash
+USERNAME=$(juju run data-integrator/0 get-credentials --format=json | yq .data-integrator/0.results.kafka.username)
+PASSWORD=$(juju run data-integrator/0 get-credentials --format=json | yq .data-integrator/0.results.kafka.password)
+```
+<!-- 
+Now, let's create a new Kubernetes service account for us to use for streaming example. As earlier, we will use `spark-client` snap to create the service account.
+
+```bash
+kubectl create namespace spark-streaming
+
+spark-client.service-account-registry create \
+  --username spark --namespace spark-streaming \
+  --properties-file ./confs/s3.conf
+``` -->
+
+Now, we can submit a job that uses the Spark streaming APIs using `spark-client.spark-submit`. The Python script named `spark_streaming.py` is available in our local MinIO S3 bucket which we created in the earlier section. The script looks like the following:
+
```

Comment:
**suggestion** Maybe here, always in the sake of being educational and build on top of what we did before, we can describe and breaking down a bit more the script, focus on the snippet for reading data instead of S3 from Kafka. After that, the API and the calculations are similar as what we presented before, where we do some group-by and count.